script.js

```diff
@@ -0,0 +1,138 @@
+let currencyRates = {
+    timestamp: 1519296206,
+    base: "EUR",
+    date: "2021-03-17",
+    rates: {
+        USD: 1.23396
+    }
+};
+
+function insertCurrencyRate() {
+    const base = document.getElementById('base').value.toUpperCase();
+    const targetCurrency = document.getElementById('targetCurrency').value.toUpperCase();
+    const rate = parseFloat(document.getElementById('rate').value);
+    
+
+    if (!currencyRates) {
+        currencyRates = {
+            timestamp: Date.now(),
+            base: base,
+            date: new Date().toISOString().split('T')[0],
+            rates: {}
+        };
+    }
+
+    if (currencyRates.base !== base) {
+        alert('Base currency mismatch! Current base is ' + currencyRates.base);
+        return;
+    }
+
+
+    currencyRates.timestamp = Date.now();
+    currencyRates.date = new Date().toISOString().split('T')[0];
+    currencyRates.rates[targetCurrency] = rate;
+    document.getElementById('newRateOutput').textContent = JSON.stringify(currencyRates, null, 2);
+    populateCurrencyDropdown();
+}
+
+function populateCurrencyDropdown() {
+    const selectedFromCurrencyLabel = document.getElementById('selected-from-currency-label');
+    const selectedToCurrencyLabel = document.getElementById('selected-to-currency-label');
+    const selectedCurrencyLabel = document.getElementById('selected-currency-label');
+    const fromCurrency = document.getElementById('fromCurrency');
+    const selectCurrency = document.getElementById('selectCurrency');
+    const toCurrency = document.getElementById('toCurrency');
+    const rates =  currencyRates.rates;
+    
+    while (fromCurrency.options.length > 1) {
```

Comment:
I don't understand what you want to do with this `while` loop. Can you clarify it, please?

Reply:
@vdaguenet  , to avoid the duplicate value in dropdowns.

Reply:
I see. So you remove all the `<option>` elements and repopulate them afterwards. It would be more optimal to check if the `<option>` for a given rate exists before adding it. In pseudo code it would be:
```js
for (const [key, value] of Object.entries(rates)) {
  if (hasOptionForRate(key) === false) {
    createOptionForRate(key, value);
  }
}
```
 It is fine to merge the current PR without this optimisation. You can add it in your PR for the week 2 assignment.